sklearn/utils/class_weight.py

```diff
@@ -41,12 +41,14 @@ def compute_class_weight(class_weight, classes, y):
     # Import error caused by circular imports.
     from ..preprocessing import LabelEncoder
 
+    _dtype = y.dtype
+
     if set(y) - set(classes):
         raise ValueError("classes should include all valid labels that can "
                          "be in y")
     if class_weight is None or len(class_weight) == 0:
         # uniform class weights
-        weight = np.ones(classes.shape[0], dtype=np.float64, order='C')
+        weight = np.ones(classes.shape[0], dtype=_dtype, order='C')
```

Comment:
This can be the cause of the following Errors, in travis:
 ```
...
ValueError: Estimator LogisticRegression raises error if 'sample_weight' parameter is of type pandas.Series
 ...
ValueError: Buffer dtype mismatch, expected 'float64_t' but got 'long'
...
```
What I don't understand is why I get no errors when I run it on my computer: 
```
 $ nosetests sklearn
...
-------------
Ran 7984 tests in 283.040s

OK (SKIP=15)
```